cpp/src/table/row_operators.cu

```diff
@@ -253,6 +259,29 @@ auto decompose_structs(table_view table,
                          std::move(verticalized_col_depths));
 }
 
+/*
+ * This helper function generates dremel data for any list-type columns in a
+ * table. This data is necessary for lexicographic comparisons.
+ */
+auto list_lex_preprocess(table_view table, rmm::cuda_stream_view stream)
+{
+  std::vector<detail::dremel_data> dremel_data;
+  std::vector<detail::dremel_device_view> dremel_device_views;
+  for (auto const& col : table) {
+    if (col.type().id() == type_id::LIST) {
+      dremel_data.push_back(detail::get_dremel_data(col, {}, stream));
+      dremel_device_views.push_back(dremel_data.back());
+    } else {
+      // TODO: Note that this constructs a device view that is in an invalid
+      // state, i.e. dereferencing any of its pointer members will lead to a
+      // seg fault. We may instead wish to create a vector of optionals.
+      dremel_device_views.emplace_back();
```

Comment:
⚠️ This feels like a fairly important TODO to fix before merging.

Reply:
I added this TODO half an hour ago when I realized what a significant hole this was. It's part of the same discussion that has now come up on [two other threads around optionals](https://github.com/rapidsai/cudf/pull/11129#discussion_r932539303), and is essentially the same problem: how does the code handle columns that aren't list columns (with 0 list columns as the extreme case but not anything categorically different).

Reply:
OK this is resolved now. We no longer ever construct these.